### PR TITLE
OpInfo for `nn.functional.cross_entropy`

### PR DESCRIPTION
**Pull request by krshrimali**
Reference: https://github.com/facebookresearch/functorch/issues/78 and https://github.com/pytorch/pytorch/issues/54261

TODOs:

* [ ] Investigate autograd failures.
* [ ] Clean up `test_nn.py` for `cross_entropy`.

cc: @mruberry @zou3519 

cc @albanD @mruberry @jbschlosser @walterddr

### CODE REVIEW

**Comment by pmeier** [comment]
@zou3519 The [failing tests](test_fn_grad_nn_functional_cross_entropy_cuda_float64) here look awful lot like the ones in #63854. Given that `cross_entropy(input)` is basically `nll_loss(log_softmax(input))`, we should fix the other PR first. Afterwards we can apply the same treatment here and see if the failures persist.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -4213,6 +4213,53 @@ def sample_inputs_to_sparse(op_info, device, dtype, requires_grad, **kwargs):
...
+        (shape, dict(ignore_index=1)),
+        (shape, dict(reduction="sum")),
+        (shape, dict(reduction="none")),
+    ]
+
+    sample_inputs = []
+    for (input_shape, kwargs), probabilities_target in itertools.product(input_shape_and_kwargs, (False, True)):
+        # FIXME:
+        if "weight" in kwargs and not probabilities_target:
+            continue
```
Comment (pmeier) [inline torch/testing/_internal/common_methods_invocations.py:4234]: On CPU this combination is causing gradgradcheck errors. I'm working on a MWE and will open an issue,

Comment (pmeier) [reply torch/testing/_internal/common_methods_invocations.py:4234]: See #64256.

**Review by zou3519 — COMMENTED** [review]
This looks pretty good. I had one suggestion for improving the test case, please let me know what you think

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -8446,6 +8502,32 @@ def ref_pairwise_distance(input1, input2):
...
+                toleranceOverride({torch.float32: tol(atol=1e-5, rtol=1e-3)}),
+                "TestJit",
+                "test_variant_consistency_jit",
+                device_type="cpu",
+            ),
+        ),
+        skips=(
+            # AssertionError: False is not true : Scalars failed to compare as equal! 0 != 1536
+            # test_ops.TestJitCUDA.test_variant_consistency_jit_nn_functional_cross_entropy_cuda_float32 leaked
+            # 1536 bytes CUDA memory on device 0
```
Comment (zou3519) [inline torch/testing/_internal/common_methods_invocations.py:8522]: Very interesting

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**